src/EnergyPlus/GeneralRoutines.cc

```diff
@@ -2154,4 +2154,47 @@ void TestReturnAirPathIntegrity(OutputFiles &outputFiles, bool &ErrFound, Array2
     }
 }
 
+void CalcTotalSensibleLatentOutput(Real64 const MassFlow,  // air mass flow rate, {kg/s}
+                                   Real64 const TDB2,      // dry-bulb temperature at state 2 {C}
+                                   Real64 const dW2,       // humidity ratio at  at state 2
+                                   Real64 const TDB1,      // dry-bulb temperature at  at state 1 {C}
+                                   Real64 const dW1,       // humidity ratio  at state 1
+                                   Real64 &TotalOutput,    // total = sensible + latent putput rate (state 2 -> State 1), {W}
+                                   Real64 &SensibleOutput, // sensible output rate (state 2 -> State 1), {W}
+                                   Real64 &LatentOutput    // latent output rate (state 2 -> State 1), {W}
+)
+{
+
+    // Purpose:
+    // returns total, sensible and latent heat rate of change of moist air transitioning 
+    // between two states. The mosit air transition can be cooling and heating process
+    // in a cooling and heating coils, or moist air transfer from zone equipment outlet to
+    // zone air node condition.
+
+    // Methodology:
+    // Q_total = m_dot * (h2 - h1)
+    // Q_sensible = m_dot * Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1(TDB2, W2, TDB1, W1);
+    // or Q_sensible = m_dot * cp_moistair_average * (TDB2 - TDB1)
+    //    cp_moistair_average = Psychrometrics::PsyCpAirFnW(0.5 * (W2 + W1));
+    //    cp_moistair_average = cp_dryair + 0.5 * (W2 + W1) * cp_watervapor
+    // Q_latent = Q_total - Q_latent;
+    // or Q_latent = m_dot * hg_average * (W2 - W1);
+    //    hg_average = Psychrometrics::PsyHfgAvgFnTdb2Tdb1(TDB2, TDB1);
+
+    // reference:
+    // na
+
+    using Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1;
+    using Psychrometrics::PsyHFnTdbW;
+
+    TotalOutput = 0.0;
+    LatentOutput = 0.0;
+    SensibleOutput = 0.0;
+    if (MassFlow > 0.0) {
+        TotalOutput = MassFlow * (Psychrometrics::PsyHFnTdbW(TDB2, dW2) - Psychrometrics::PsyHFnTdbW(TDB1, dW1)); // total addition/removal rate, {W};
+        SensibleOutput = MassFlow * Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1(TDB2, dW2, TDB1, dW1); // sensible addition/removal rate, {W};
```

Comment:
@Nigusse There are many ways to split total. Since this function uses average W to calculate Cp, it is better to provide another function use min or inlet W to calculate sensible with a whole set of functions. Or, an optional argument may be used as
CalcTotalSensibleLatentOutput(....,SplitFlag)
SplitFlag = 1, average W
=2, Inlet W
= 3, min W

Reply:
@lgu1234 ? We should use only one method. We should use one function and call it from everywhere.

Reply:
@mjwitte I agree that it is better to have a single method. I also prefer to use the same method to process air movement, such as infiltration, mixing and cross mixing. The existing cooling coil calculation uses the min W to calculate the sensible. It is a good method and needs to keep. For the coiling coil, the min W occurs at the supply side, as inlet. This is the preferred method by using the source side condition theoretically.

However, the min W could occur at either source side or receiving side for infiltration, due to air movement from outdoor to indoor. Shall we use source side W, instead of min W? in addition, when we deal with mixing objects, the average W is preferred, because sensible calculations from source side and receiving side will be equal, without energy leak.

The above is my thought. That is the reason why I propose 3 methods, to allow case dependent.

Reply:
The specific heat of air changes with humidity ratio. For a cooling coil that removes moisture the specific heat used in the calculation of energy transfer, at least as far as I know, is not the entering or leaving air condition. It's the average (or the integral). For infiltration or mixing I would think the same logic applies. If there are references that say otherwise we need to review those to make sure the energy calculation is correct. It doesn't matter what calculation we use now, right or wrong, those need to be updated so we stop going in circles with this type of change. It might take more than 1 method, I'm not sure. But if 1 method can cover all cases then we should use only 1.